### ISSUE TITLE
vite_ruby missing css file from manifest when build.cssCodeSplit is false

### ISSUE THREAD

**Issue by chambersjason** [ElMassimo/vite_ruby#109]
[troubleshooting section]: https://vite-ruby.netlify.app/guide/troubleshooting.html

- [X] I have tried upgrading by running `bundle update vite_ruby`.
- [X] I have read the __[troubleshooting section]__ before opening an issue.

### Description 📖

with the following `vite.config.ts` im running into an issue where the single css file output from rollup isn't making its way into the `manifest.json` file

```typescript
export default defineConfig({
  build: {
    cssCodeSplit: false,
  },
  plugins: [reactRefresh(), RubyPlugin()],
  resolve: {
    alias: {
      src: resolve(__dirname, 'src'),
    },
  },
});
```

when `build.cssCodeSplit` is `true`, the css files are correctly output to the generated manifest file

```json
{
  "classic_taos_interop.tsx": {
    "file": "/vite/assets/classic_taos_interop.caa092eb.js",
    "src": "classic_taos_interop.tsx",
    "isEntry": true,
    "imports": [
      {
        "file": "/vite/assets/vendor.a1716101.js"
      },
      {
        "file": "/vite/assets/duo_mfa.51620be2.js",
        "imports": [
          {
            "file": "/vite/assets/vendor.a1716101.js"
          }
        ],
        "css": [
          "/vite/assets/duo_mfa.f69369a9.css"
        ],
        "assets": [
          "assets/close.6dbfea5e.svg"
        ]
      }
    ]
  },
  "main.tsx": {
    "file": "/vite/assets/main.b0644003.js",
    "src": "main.tsx",
    "isEntry": true,
    "imports": [
      {
        "file": "/vite/assets/vendor.a1716101.js"
      },
      {
        "file": "/vite/assets/duo_mfa.51620be2.js",
        "imports": [
          {
            "file": "/vite/assets/vendor.a1716101.js"
          }
        ],
        "css": [
          "/vite/assets/duo_mfa.f69369a9.css"
        ],
        "assets": [
          "assets/close.6dbfea5e.svg"
        ]
      }
    ],
    "css": [
      "/vite/assets/main.d32c6a89.css"
    ],
    "assets": [
      "assets/taos.e8de4759.jpg"
    ]
  },
  "_vendor.a1716101.js": {
    "file": "/vite/assets/vendor.a1716101.js"
  },
  "_duo_mfa.51620be2.js": {
    "file": "/vite/assets/duo_mfa.51620be2.js",
    "imports": [
      {
        "file": "/vite/assets/vendor.a1716101.js"
      }
    ],
    "css": [
      "/vite/assets/duo_mfa.f69369a9.css"
    ],
    "assets": [
      "assets/close.6dbfea5e.svg"
    ]
  }
}
```

when `build.cssCodeSplit` is `false`, the terminal output shows the `styles.css` file being created, but it is not anywhere in the manifest

```
rendering chunks...
../../public/vite/assets/close.6dbfea5e.svg                 1.21kb
../../public/vite/assets/taos.e8de4759.jpg                  15.75kb
../../public/vite/manifest-assets.json                      0.00kb
../../public/vite/manifest.json                             0.71kb
../../public/vite/assets/classic_taos_interop.c0bfecd9.js   1.89kb / brotli: 0.74kb
../../public/vite/assets/classic_taos_interop.c0bfecd9.js.map 8.42kb
../../public/vite/assets/style.0c577cb2.css                 21.96kb / brotli: 3.64kb
../../public/vite/assets/duo_mfa.f66d223f.js                36.33kb / brotli: 9.64kb
../../public/vite/assets/duo_mfa.f66d223f.js.map            149.38kb
../../public/vite/assets/main.007354ee.js                   124.29kb / brotli: 26.18kb
../../public/vite/assets/main.007354ee.js.map               391.89kb
../../public/vite/assets/vendor.3f056704.js                 302.28kb / brotli: 79.16kb
../../public/vite/assets/vendor.3f056704.js.map             1199.71kb
```

```json
{
  "classic_taos_interop.tsx": {
    "file": "/vite/assets/classic_taos_interop.c0bfecd9.js",
    "src": "classic_taos_interop.tsx",
    "isEntry": true,
    "imports": [
      {
        "file": "/vite/assets/vendor.3f056704.js"
      },
      {
        "file": "/vite/assets/duo_mfa.f66d223f.js",
        "imports": [
          {
            "file": "/vite/assets/vendor.3f056704.js"
          }
        ],
        "assets": [
          "assets/close.6dbfea5e.svg"
        ]
      }
    ]
  },
  "main.tsx": {
    "file": "/vite/assets/main.007354ee.js",
    "src": "main.tsx",
    "isEntry": true,
    "imports": [
      {
        "file": "/vite/assets/vendor.3f056704.js"
      },
      {
        "file": "/vite/assets/duo_mfa.f66d223f.js",
        "imports": [
          {
            "file": "/vite/assets/vendor.3f056704.js"
          }
        ],
        "assets": [
          "assets/close.6dbfea5e.svg"
        ]
      }
    ],
    "assets": [
      "assets/taos.e8de4759.jpg"
    ]
  },
  "_vendor.3f056704.js": {
    "file": "/vite/assets/vendor.3f056704.js"
  },
  "_duo_mfa.f66d223f.js": {
    "file": "/vite/assets/duo_mfa.f66d223f.js",
    "imports": [
      {
        "file": "/vite/assets/vendor.3f056704.js"
      }
    ],
    "assets": [
      "assets/close.6dbfea5e.svg"
    ]
  }
}
```

i added some logging to my local `vite-plugin-ruby` installation, and i see the style file being in the rollup bundle within the `generateBundle` function 

https://github.com/ElMassimo/vite_ruby/blob/dd4ac9034fc335f7ebd055af14067aa40ca1562f/vite-plugin-ruby/src/manifest.ts#L77-L88

but it doesnt seem to end up in the manifest used by `vite_rails`. and the `manifest-assets.json` file is just an empty object `{}`.

rollup bundle output
```
...
  'assets/style.0c577cb2.css': {
    fileName: 'assets/style.0c577cb2.css',
    isAsset: [Getter],
    name: 'style.css',
    source: '[i omitted this]'
    type: 'asset'
  },
...
```

i noticed theres some filtering for `asset` chunks in `extractChunkStylesheets`, so im wondering if the source of the problem is around there

https://github.com/ElMassimo/vite_ruby/blob/dd4ac9034fc335f7ebd055af14067aa40ca1562f/vite-plugin-ruby/src/manifest.ts#L33-L47

### Reproduction 🐞

im hoping i provided enough info to help solve. if you need more info though i can spend some time trying to set something up to repro

<details>
<summary>Vite Ruby Info</summary>

_Run `bin/rake vite:info` and provide the output:_

```
bin/vite present?: true
vite_ruby: 1.2.18
vite_rails: 2.0.12
rails: 6.0.3.3
node: v12.15.0
npm: 7.9.0
yarn: 1.22.4
pnpm: 
ruby: jruby 9.2.14.0 (2.5.7) 2020-12-08 ebe64bafb9 OpenJDK 64-Bit Server VM 11.0.10+9 on 11.0.10+9 +jit [darwin-x86_64]

installed packages:
taos@0.0.1 [removed]taos
├─┬ vite-plugin-ruby@2.0.4
│ └── vite@2.4.3 deduped
└── vite@2.4.3

```
</details>

### Logs 📜

_If not providing a reproduction:_

<details>
<summary>Output</summary>

_Run `DEBUG=vite-plugin-ruby:* bin/vite dev` or `DEBUG=vite-plugin-ruby:* bin/vite build` and provide the output:_

```
▶ DEBUG=vite-plugin-ruby:* bin/vite build
Building with Vite ⚡️
vite v2.4.3 building for production...
transforming...
✓ 594 modules transformed.
rendering chunks...
{
  fileName: 'assets/close.6dbfea5e.svg',
  isAsset: [Getter],
  name: undefined,
  source: <Buffer 3c 73 76 67 20 76 65 72 73 69 6f 6e 3d 22 31 2e 31 22 20 69 64 3d 22 43 61 70 61 5f 31 22 20 78 6d 6c 6e 73 3d 22 68 74 74 70 3a 2f 2f 77 77 77 2e 77 ... 1194 more bytes>,
  type: 'asset'
}
{
  fileName: 'assets/taos.e8de4759.jpg',
  isAsset: [Getter],
  name: undefined,
  source: <Buffer ff d8 ff e0 00 10 4a 46 49 46 00 01 01 00 00 64 00 64 00 00 ff e1 00 88 45 78 69 66 00 00 4d 4d 00 2a 00 00 00 08 00 05 01 12 00 03 00 00 00 01 00 01 ... 16074 more bytes>,
  type: 'asset'
}
{
  fileName: 'assets/style.0c577cb2.css',
  isAsset: [Getter],
  name: 'style.css',
  source: '[removed]',
  type: 'asset'
}
../../public/vite/assets/close.6dbfea5e.svg                 1.21kb
../../public/vite/assets/taos.e8de4759.jpg                  15.75kb
../../public/vite/manifest-assets.json                      0.00kb
../../public/vite/manifest.json                             0.71kb
../../public/vite/assets/classic_taos_interop.c0bfecd9.js   1.89kb / brotli: 0.74kb
../../public/vite/assets/classic_taos_interop.c0bfecd9.js.map 8.42kb
../../public/vite/assets/style.0c577cb2.css                 21.96kb / brotli: 3.64kb
../../public/vite/assets/duo_mfa.f66d223f.js                36.33kb / brotli: 9.64kb
../../public/vite/assets/duo_mfa.f66d223f.js.map            149.38kb
../../public/vite/assets/main.007354ee.js                   124.29kb / brotli: 26.18kb
../../public/vite/assets/main.007354ee.js.map               391.89kb
../../public/vite/assets/vendor.3f056704.js                 302.28kb / brotli: 79.16kb
../../public/vite/assets/vendor.3f056704.js.map             1199.71kb
2021-07-29T05:58:37.867Z vite-plugin-ruby:config {
  base: '/vite/',
  build: {
    emptyOutDir: true,
    sourcemap: true,
    cssCodeSplit: false,
    assetsDir: 'assets',
    manifest: true,
    outDir: '../../public/vite',
    rollupOptions: { input: [Object], output: [Object] }
  },
  root: '[removed]taos/src/entrypoints',
  server: {
    host: 'localhost',
    https: false,
    port: 3036,
    strictPort: true,
    fsServe: { root: '[removed]taos' }
  },
  entrypoints: {
    classic_taos_interop: '[removed]taos/src/entrypoints/classic_taos_interop.tsx',
    main: '[removed]taos/src/entrypoints/main.tsx'
  }
}
2021-07-29T05:58:44.011Z vite-plugin-ruby:assets-manifest { manifest: Map {} }

```
</details>

### Screenshots 📷

n/a

**Comment by ElMassimo** [comment]
Thanks for the report Jason.

I've released [`vite-plugin-ruby@2.0.5`](https://github.com/ElMassimo/vite_ruby/releases/tag/vite-plugin-ruby%402.0.5) with a patch.

<hr/>

Have in mind that you'll need to manually inject the generated `style.css` in your app, but only when the Vite dev server is not running. Something like this should do the trick:

```erb
<%= vite_stylesheet_tag 'style' unless ViteRuby.instance.dev_server_running? %>
```

**Comment by chambersjason** [comment]
@ElMassimo thanks much for the quick turnaround!
